flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/physical/batch/PushLocalHashAggWithCalcIntoScanRule.java

```diff
@@ -85,7 +82,11 @@ public void onMatch(RelOptRuleCall call) {
         BatchPhysicalCalc calc = call.rel(2);
         BatchPhysicalTableSourceScan oldScan = call.rel(3);
 
-        int[] calcRefFields = getRefFiledIndex(calc);
+        // For count(*) and count(n) we need to ignore the calc.
+        int[] calcRefFields = null;
```

Comment:
Can these codes also be moved into `pushLocalAggregateIntoScan`  because of duplication?